src/components/victory-bar/helper-methods.js

```diff
@@ -104,10 +102,9 @@ module.exports = {
     } else if (stringMap.x) {
       return (datum.x - 1);
     } else {
-      const allX = datasets.map((dataset) => {
-        return dataset.data.map((d) => d.x);
-      });
-      const uniqueX = uniq(flatten(allX));
+      const uniqueX = uniq(datasets.reduce((prev, dataset) => {
```

Comment:
@boygirl if you move this to its own function (and memoize it) then the performance of renderBars() in https://github.com/FormidableLabs/victory-chart/blob/b8eb4cad0feca11bdb51eb94d006921cc6cf9925/src/components/victory-bar/victory-bar.jsx#L285 will be improved significantly, since for every datum in dataset.data you're recomputing uniqueX even though it doesn't depend on datum, the only thing that depends on datum is the index (so either store uniqueX outside the map in renderBars() or memoize getUniqueX() and you only compute it once